tests/ui/unnecessary_filter_map.rs

```diff
@@ -21,6 +22,8 @@ fn main() {
     //~^ ERROR: this `.filter_map` can be written more simply using `.map`
 
     let _ = (0..4).filter_map(i32::checked_abs);
+
+    let _ = vec![Some(10), None].into_iter().filter_map(|x| Some(x));
```

Comment:
The linked issue also mentions passing a path to `Some`, so maybe we should add this as a test too

```rust
let _ = (0..4).filter_map(Some);
```

Reply:
The more tests the merrier 😎